homeassistant/components/fritzbox/config_flow.py

```diff
@@ -0,0 +1,135 @@
+"""Config flow for AVM Fritz!Box."""
+import socket
+from urllib.parse import urlparse
+
+from pyfritzhome import Fritzhome, LoginError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.ssdp import ATTR_SSDP_LOCATION, ATTR_UPNP_FRIENDLY_NAME
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+
+# pylint:disable=unused-import
+from .const import DOMAIN
+
+DEFAULT_HOST = "fritz.box"
+DATA_SCHEMA_USER = vol.Schema(
+    {
+        vol.Required(CONF_HOST, default=DEFAULT_HOST): str,
+        vol.Required(CONF_USERNAME): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+DATA_SCHEMA_CONFIRM = vol.Schema(
+    {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+)
+
+RESULT_AUTH_MISSING = "auth_missing"
+RESULT_SUCCESS = "success"
+RESULT_NOT_FOUND = "not_found"
+
+
+def _get_ip(host):
+    if host is None:
```

Comment:
We should check that host isn't `None` before passing it to this function.

Reply:
Do we need this wrapper at all? I would say let's call `socket.gethostbyname` directly. If `host` is `None` we've more problems than the failing IP.